code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,1381 @@
+		var/list/choices = list()
+		for(var/mob/living/carbon/human/H in get_turf(src))
+			if((H.stat == DEAD) && (!ismonkeybasic(H))) // Can only absorb dead humans.
+				choices += H
+//				to_chat(src, "<span class='warning'>A thing can be absorbed</span>") Debug thing
+		absorb_target = input(src,"What do you wish to absorb?", name) as null|mob in choices
+
+	if((absorb_target) && (absorbing == 0)) // We have selected a target.
+		var/mob/living/carbon/human/H = absorb_target
+		if(H in get_turf(src)) // Don't even try if you aren't in the same turf to begin with
+			dir = 1
+			canmove = 0
+			absorbing = 1
+			flick("thing_absorbin",src)
+			icon_state = "thing_absorbin"
+			playsound(src, 'sound/thing/savor.ogg', 100, 1)
+			visible_message("<span class='warning'>\the [src] kneels down, savoring \the [H]...</span>")
+			deathmessage = "cannot die... Not yet!"
+			sleep(45)  // Long time to prevent the action from being completed, or to shoot it.
+			if(H in get_turf(src)) // Check to see if the target is in the same turf as the thing. If not, see the else condition below.
+				anchored = 1  // Point of no return, both the thing and the target are anchored and the darkness quickly forms.
+				H.anchored = 1
+				flick("thing_absorbin2",src)
+				sleep(10.25)
+				icon_state = "thing_absorbing"
+				var/absorbckey = H.ckey
+				sleep(29.4) // Bit of time to damage the thing. Enough time to play the animation six times.
+				H.add_splatter_floor()
+				qdel(absorb_target)
+				if(can_convert == 1)
+					var/turf/B = get_turf(src)
+					var/atom/movable/overlay/animation = new /atom/movable/overlay(B)
+					animation.name = "???"
+					animation.density = 0
+					animation.anchored = 1
+					animation.icon = 'icons/mob/mob.dmi'
+					animation.icon_state = "serf_appear"
+					animation.layer = 4
+					animation.desc = "What..."
+					animation.master = holder
+					flick("serf_appear",animation)
+					sleep(14)
+					if(animation)
+						qdel(animation)
+						var/mob/living/simple_animal/hostile/serf/S = new /mob/living/simple_animal/hostile/serf(loc)
+						S.ckey = absorbckey
+						to_chat(S, "<span class='danger'>You are now a chaos bringer! Follow your master to victory. The longer you survive, the stronger you become.</span>")
+						S.dir = 2
+						S.visible_message("<B>A strange being rises...</b>")
+				flick("thing_absorbout",src)
+				playsound(src, 'sound/thing/absorb.ogg', 80, 1)
+				visible_message("<B>The tentacles are absorbed in a sickening crunch, and so is \the [H]...</b>")
+				layer = 4.1
+				sleep(11.8)
+				dir = 1
+				icon_state = "thing"
+				anchored = 0
+				canmove = 1
+				absorb_target = null
+				absorbckey = null
+				absorbing = 0
+				points++ // Get a point
+				absorptions++
+				layer = 4
+				if(health < maxHealth)
+					if(health == 0)
+						adjustBruteLoss(-(maxHealth/10)) // The absorption can't be stopped. The thing will be revived if it dies, although it will have very low health.
+						stat = 0
```

Comment:
VERY BAD
If you decide to continue this route, use `update_revive`, otherwise you're going to split the state machine in half, they're going to see deadchat and other such badness

Reply:
Where would I put `update_revive`?